### PR TITLE
Pin knative/pkg to 1.8

### PR DESCRIPTION
**Pull request by vdemeester**
# Changes

This pin the dependency to the 1.8 release of knative instead of a
commit in main.

The commit (https://github.com/knative/pkg/commit/8178c38717851ad42b96e93f86169af59f5e92b4)
that #5643 required is available in 1.8 (see https://github.com/knative/pkg/commits/release-1.8)

Signed-off-by: Vincent Demeester <vdemeest@redhat.com>

/kind misc

# Submitter Checklist

As the author of this PR, please check off the items in this checklist:

- [ ] Has [Docs](https://github.com/tektoncd/community/blob/main/standards.md#docs) included if any changes are user facing
- [ ] Has [Tests](https://github.com/tektoncd/community/blob/main/standards.md#tests) included if any functionality added or changed
- [x] Follows the [commit message standard](https://github.com/tektoncd/community/blob/main/standards.md#commits)
- [x] Meets the [Tekton contributor standards](https://github.com/tektoncd/community/blob/main/standards.md) (including
  functionality, content, code)
- [x] Has a kind label. You can add one by adding a comment on this PR that contains `/kind <type>`. Valid types are bug, cleanup, design, documentation, feature, flake, misc, question, tep
- [ ] Release notes block below has been updated with any user facing changes (API changes, bug fixes, changes requiring upgrade notices or deprecation warnings)
- [ ] Release notes contains the string "action required" if the change requires additional action from users switching to the new release

# Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by vdemeester** [comment]
cc @afrittoli @imjasonh @abayer

File: go.mod [hunk 1]
```diff
@@ -34,7 +34,7 @@ require (
 	k8s.io/code-generator v0.25.2
 	k8s.io/klog v1.0.0
 	k8s.io/kube-openapi v0.0.0-20220803162953-67bda5d908f1
-	knative.dev/pkg v0.0.0-20221014164553-b812affa3893
+	knative.dev/pkg v0.0.0-20221011175852-714b7630a836
```
Comment (afrittoli) [inline go.mod:37]: NIT: It would be nice to have a comment in here so we know it's 1.8?

**Review by afrittoli — COMMENTED** [review]
Being on a specific release makes it easier to pull in patch releases, so +1 on this.

/approve

**Comment by lbernick** [comment]
> Being on a specific release makes it easier to pull in patch releases, so +1 on this.

What's the reason this is easier out of curiosity? also does [this line](https://github.com/tektoncd/pipeline/blob/b63c1e28944d0c27dba964bbf87902253ba82b0e/hack/update-deps.sh#L25) need to be updated?

**Comment by vdemeester** [comment]
> > Being on a specific release makes it easier to pull in patch releases, so +1 on this.
> 
> What's the reason this is easier out of curiosity? also does [this line](https://github.com/tektoncd/pipeline/blob/b63c1e28944d0c27dba964bbf87902253ba82b0e/hack/update-deps.sh#L25) need to be updated?

Oh very good point, it needs to.. I forgot to commit that file.. 🤦🏼

**Pull request merged**